### DISCUSSION TITLE
require() of ES Module /node_modules/xxx/index.js/ from  xxx.js not supported. Instead change the require of index.js

### DISCUSSION THREAD

**Discussion by peterroe** [nuxt/nuxt#20693] [General]
When using @antv/g2@next, I encountered the following error report：
```
require() of ES Module /home/projects/github-ewxpgt/node_modules/.pnpm/d3-hierarchy@3.1.2/node_modules/d3-hierarchy/src/index.js from /home/projects/github-ewxpgt/node_modules/.pnpm/@antv+g2@5.0.5_@antv+g-lite@1.0.54/node_modules/@antv/g2/lib/mark/utils.js not supported. Instead change the require of index.js in /home/projects/github-ewxpgt/node_modules/.pnpm/@antv+g2@5.0.5_@antv+g-lite@1.0.54/node_modules/@antv/g2/lib/mark/utils.js to a dynamic import() which is available in all CommonJS modules.
```
I tried to reproduce it in stackblitz, the address is here: https://stackblitz.com/edit/github-ewxpgt?file=app.vue
The normal performance should be to render a statistical graph, but in fact, the wrong result was obtained, and the operation of constantly trying to refresh the page failed to obtain the expected result

I would appreciate it if anyone could help me😊😊😊

**Comment by AloisSeckar** [comment]
The root of your problem is that packages from `d3` family were marked as "ESM only", so they can only be used through `import` keyword, not `require()` function. And some 3rd party package your're trying to use still does that. 

According to the error message it comes from `@antv/g-lite` version 1.0.54. This is somehow wired inside `@antv/g2` (in `pnpm-lock.yml` there is `version: 5.0.5(@antv/g-lite@1.0.54)`. I am not quite sure if this can be overriden somehow (I tried but failed), nor if it helps.

**Yet, I got somewhere by downgrading `@antv/g2` to version `4.2.10`.**

However, because there is a similar issue with another d3 package, I had to add following override into `package.json`:
```
  "pnpm": {
    "overrides": {
      "d3-interpolate": "2.0.1"
    }
  }
```

This got me through, but then I also had to change the implementation of the graph construction because apparently v4 [used another syntax](https://github.com/antvis/G2/issues/4658). Those steps left me error-free and with the graph being rendered.

Here is the StackBlitz repo with my changes:
https://stackblitz.com/edit/github-ewxpgt-r4hgf7?file=app.vue
